Index: DynamicLoader.cxx
===================================================================
RCS file: /cvsroot/ParaView3/ParaView3/VTK/Utilities/kwsys/DynamicLoader.cxx,v
retrieving revision 1.26
diff -u -r1.26 DynamicLoader.cxx
--- DynamicLoader.cxx   28 Sep 2009 15:37:22 -0000      1.26
+++ DynamicLoader.cxx   7 Mar 2010 00:32:01 -0000
@@ -444,7 +444,7 @@
 // 5. Implementation for systems without dynamic libs
 // __gnu_blrts__ is IBM BlueGene/L
 // __LIBCATAMOUNT__ is defined on Catamount on Cray compute nodes
-#if defined(__gnu_blrts__) || defined(__LIBCATAMOUNT__)
+#if defined(__gnu_blrts__) || defined(__LIBCATAMOUNT__) || defined(__CRAYXT_COMPUTE_LINUX_TARGET)
 #include <string.h> // for strerror()
 #define DYNAMICLOADER_DEFINED 1
