Index: Qt/Components/pq3DWidget.cxx
===================================================================
RCS file: /cvsroot/ParaView3/ParaView3/Qt/Components/pq3DWidget.cxx,v
retrieving revision 1.38
diff -u -r1.38 pq3DWidget.cxx
--- Qt/Components/pq3DWidget.cxx	16 Jul 2009 14:20:03 -0000	1.38
+++ Qt/Components/pq3DWidget.cxx	24 Jan 2010 20:42:08 -0000
@@ -406,16 +406,13 @@
     }
 
   vtkSMProxy* pxy = this->proxy();
-  unsigned int max = hints->GetNumberOfNestedElements();
-  for (unsigned int cc=0; cc < max; cc++)
+  unsigned int max_props = hints->GetNumberOfNestedElements();
+  for (unsigned int i=0; i < max_props; i++)
     {
-    unsigned int max_props = hints->GetNumberOfNestedElements();
-    for (unsigned int i=0; i < max_props; i++)
-      {
-      vtkPVXMLElement* propElem = hints->GetNestedElement(i);
-      this->setControlledProperty(propElem->GetAttribute("function"),
+    vtkPVXMLElement* propElem = hints->GetNestedElement(i);
+    this->setControlledProperty(
+        propElem->GetAttribute("function"),
         pxy->GetProperty(propElem->GetAttribute("name")));
-      }
     }
 }
 