--- /share/sge6.2/default/pe_scripts/tacc_vglrun        2009-11-12 13:51:35.000000000 -0600
+++ apps/pv_vglrun      2010-06-08 16:55:21.000000000 -0500
@@ -57,10 +57,8 @@
   export TACC_DISPLAY=${GL_DISPLAY[0]}
 fi

-# If we're on a slave node, point back to the master node
-if [ x$HOSTNAME != x`hostname` ]; then
-    export VGL_CLIENT=$HOSTNAME
-    export DISPLAY=${HOSTNAME}${DISPLAY}
-fi
+export VGL_CLIENT=`hostname`
+export DISPLAY=${TACC_DISPLAY}

 vglrun -d $TACC_DISPLAY $*
+
