Index: Servers/Executables/CMakeLists.txt
===================================================================
RCS file: /cvsroot/ParaView3/ParaView3/Servers/Executables/CMakeLists.txt,v
retrieving revision 1.21
diff -u -r1.21 CMakeLists.txt
--- Servers/Executables/CMakeLists.txt  7 Apr 2010 18:17:27 -0000       1.21
+++ Servers/Executables/CMakeLists.txt  23 Apr 2010 20:35:31 -0000
@@ -23,7 +23,11 @@
 FOREACH(pvexe ${PV_EXE_LIST})
   add_executable_with_forwarding(pv_exe_suffix
     ${pvexe} ${pvexe}.cxx)
-  TARGET_LINK_LIBRARIES(${pvexe}${pv_exe_suffix} ${LIBS})
+  TARGET_LINK_LIBRARIES(
+    ${pvexe}${pv_exe_suffix}
+    ${LIBS}
+    /nics/c/home/bloring/apps/PV-Plugins/SVTK/gnu/R/libSciVisToolKit.a
+    /nics/c/home/bloring/apps/PV-Plugins/VPIC/gnu/R/libVPICReader.a)
   # Install exe
   IF(NOT PV_INSTALL_NO_RUNTIME)
     INSTALL(TARGETS ${pvexe}
Index: Servers/Executables/pvserver.cxx
===================================================================
RCS file: /cvsroot/ParaView3/ParaView3/Servers/Executables/pvserver.cxx,v
retrieving revision 1.27
diff -u -r1.27 pvserver.cxx
--- Servers/Executables/pvserver.cxx    24 Oct 2008 15:17:12 -0000      1.27
+++ Servers/Executables/pvserver.cxx    23 Apr 2010 20:35:31 -0000
@@ -20,10 +20,15 @@
 #include "vtkSMApplication.h"
 #include "vtkSMProperty.h"
 #include "vtkToolkits.h" // For VTK_USE_MPI
+#include "vtkPVPlugin.h"

 // forward declare the initialize function
 static void ParaViewInitializeInterpreter(vtkProcessModule* pm);

+
+PV_PLUGIN_IMPORT_INIT(SciVisToolKit)
+PV_PLUGIN_IMPORT_INIT(VPICReader)
+
 //----------------------------------------------------------------------------
 int main(int argc, char* argv[])
 {
@@ -42,6 +47,10 @@
     vtkSMApplication* smapplication = vtkSMApplication::New();
     vtkSMProperty::SetCheckDomains(0);
     smapplication->Initialize();
+
+    PV_PLUGIN_IMPORT(SciVisToolKit)
+    PV_PLUGIN_IMPORT(VPICReader)
+
     ret = pvmain->Run(options);
     smapplication->Finalize();
     smapplication->Delete();
