Index: Qt/Components/pqObjectPanelImplementation.cxx.in
===================================================================
RCS file: /cvsroot/ParaView3/ParaView3/Qt/Components/pqObjectPanelImplementation.cxx.in,v
retrieving revision 1.5
diff -u -r1.5 pqObjectPanelImplementation.cxx.in
--- Qt/Components/pqObjectPanelImplementation.cxx.in	19 Feb 2009 19:40:11 -0000	1.5
+++ Qt/Components/pqObjectPanelImplementation.cxx.in	18 Dec 2009 22:42:53 -0000
@@ -50,16 +50,18 @@
 
 bool @PANEL_NAME@Implementation::canCreatePanel(pqProxy* proxy) const
 {
-  if (QString("@PANEL_XML_GROUP@") != proxy->getProxy()->GetXMLGroup())
+  foreach (QString group, QString("@PANEL_XML_GROUP@").split(';',QString::SkipEmptyParts))
     {
-    return false;
-    }
-  foreach (QString name, QString("@PANEL_XML_NAME@").split(';',
-      QString::SkipEmptyParts))
-    {
-    if (name == proxy->getProxy()->GetXMLName())
+    if (group != proxy->getProxy()->GetXMLGroup())
+      {
+      continue;
+      }
+    foreach (QString name, QString("@PANEL_XML_NAME@").split(';',QString::SkipEmptyParts))
       {
-      return true;
+      if (name == proxy->getProxy()->GetXMLName())
+        {
+        return true;
+        }
       }
     }
   return false;
